### PR TITLE
Phase II.D: Handlers for ISharedMinterMerkleV0, w/Generic Event Handlers

### PR DESCRIPTION
**Pull request by ryley-o**
## Description of the change

Add handlers for IFilteredMinterMerkleV0, including adding all generic event handlers for shared minters.

Shared Merkle minters utilize the generic event handlers (e.g. when setting things like Merkle root), so the generic events are added and tested as part of this PR.

## Tests

- e2e tests are added for all events in `IFiltereMinterMerkleV0`
- e2e tests are added to verify the handling behavior of all generic events used by the new `MinterSetPriceMerkleV5` minter.
  - note: e2e tests are not added for the events not utilized by any minters, since that isn't possible without writing dummy minters. dev comments are added to highlight this in the generic event handling test file.

### CODE REVIEW

File: tests/e2e/runner/__tests__/minter-suite-v2/i-filtered-shared-merkle.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import { describe, test, expect } from "@jest/globals";
```
Comment (yoshiwarab) [inline tests/e2e/runner/__tests__/minter-suite-v2/i-filtered-shared-merkle.ts:1]: Should the filename be i-filtered-shared-merkle.test.ts

**Review by yoshiwarab — APPROVED** [review]

File: tests/e2e/runner/__tests__/minter-suite-v2/i-filtered-shared-merkle.ts (hunk 1 above)
Comment (ryley-o) [reply tests/e2e/runner/__tests__/minter-suite-v2/i-filtered-shared-merkle.ts]: yes - updated in [51081c1](https://github.com/ArtBlocks/artblocks-subgraph/pull/230/commits/51081c1c8e40302300691b05061e24ee8f9972e3)

**Comment by ryley-o** [comment]
@yoshiwarab - I added some e2e testing robustness changes in [a1d0993](https://github.com/ArtBlocks/artblocks-subgraph/pull/230/commits/a1d0993247b1d057d62df7905e377787f68f18aa).

Specifically:
- --runInBand was super important to keep Jest from running tests in separate files in parallel, resulting in non-deterministic test failures and chaos 😅 
- the timeout did need to be set to >15 seconds for some of the longer tests that perform a sequence of operate-validate-operate-validate-operate-validate, where operations are on the blockchain, and validations are queries to the subgraph.

File: src/i-shared-merkle-mapping.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+import { loadOrCreateMinter } from "./helpers";
+
+import { setMinterExtraMinterDetailsValue } from "./extra-minter-details-helpers";
+
+///////////////////////////////////////////////////////////////////////////////
+// EVENT HANDLERS start here
+///////////////////////////////////////////////////////////////////////////////
+
+export function handleDefaultMaxInvocationsPerAddress(
```
Comment (jakerockland) [inline src/i-shared-merkle-mapping.ts:14]:
```suggestion
/**
 * Handles the event that updates the default max invocations per address. 
 * Loads or creates the minter, updates the minter's details, and refreshes
 * the minter's timestamp to induce a sync.
 * @param event The event containing the updated max invocations per address.
 */
export function handleDefaultMaxInvocationsPerAddress(
```

File: src/i-shared-merkle-mapping.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    event.params.defaultMaxInvocationsPerAddress,
+    minter
+  );
+
+  // update minter's updatedAt timestamp to induce a sync, and save
+  minter.updatedAt = event.block.timestamp;
+  minter.save();
+}
+
+export function handleDelegationRegistryUpdated(
```
Comment (jakerockland) [inline src/i-shared-merkle-mapping.ts:32]:
```suggestion
/**
 * Handles the event that updates the delegation registry address. 
 * Loads or creates the minter, updates the minter's details, and refreshes
 * the minter's timestamp to induce a sync.
 * @param event The event containing the updated delegation registry address.
 */
export function handleDelegationRegistryUpdated(
```

**Review by jakerockland — APPROVED** [review]

**Pull request merged**